lib/MetaCPAN/Script/Snapshot.pm

```diff
@@ -0,0 +1,289 @@
+package MetaCPAN::Script::Snapshot;
+
+use strict;
+use warnings;
+
+use Log::Contextual qw( :log );
+
+use MetaCPAN::Types qw( Bool Int Str File );
+use Moose;
+use DateTime;
+use Try::Tiny;
+use Sys::Hostname;
+use HTTP::Tiny;
+use Cpanel::JSON::XS;
+use DDP;
+
+with 'MetaCPAN::Role::Script', 'MooseX::Getopt::Dashes';
+
+my $hostname = hostname;
+my $mode = $hostname =~ /dev/ ? 'testing' : 'production';
+
+# So we dont' break production
+my $bucket = "mc-${mode}-backups";
+
+my $repository_name = 'our_backups';
+
+## Modes
+has setup => (
+    is            => 'ro',
+    isa           => Bool,
+    documentation => 'Setup the connection with ES',
```

Comment:
Could we set explicit defaults for the `Bool` attributes?  I'm assuming they default to `false`, but I just tried to look it up and the Moose docs weren't clear to me.